prereise/gather/griddata/hifld/data_process/profiles.py

```diff
@@ -100,3 +104,98 @@ def build_wind(wind_plants, download_directory, year):
         directory=download_directory,
     )
     return profiles
+
+
+def filter_anomalies_impute(series, capacity, rel_min=-1, rel_max=2, max_count=100):
+    """Given a time-series of generation and an assumed capacity, identify anomalous
+    points and replace them by linear imputation (unless there are too many, in which
+    case return a copy of the original.
+
+    :param pandas.Series/numpy.ndarray series: 1-D data to filter and impute (MW).
+    :param int/float capacity: assumed capacity (MW).
+    :param int/float rel_min: lower screening value (relative to ``capacity``).
+    :param int/float rel_max: upper screening value (relative to ``capacity``).
+    :param int max_count: the maximum number of anomalies that can be replaced. If more
+        than this number exist, the series will be returned as is.
+    :return: (*pandas.Series*) -- data, imputed as applicable.
+    :raise ValueError: if series isn't a 1-D object
+    """
+    if len(series.shape) != 1:
+        raise ValueError("series must be of one-dimentional shape")
+    suspected_anomalies = (series > rel_max * capacity) | (series < rel_min * capacity)
+    output = series.copy()
+    if suspected_anomalies.sum() == 0:
+        return output
+    if suspected_anomalies.sum() > max_count:
+        print(f"Too many suspected anomalies for {series.name}, none will be filtered")
+        return output
+    output.loc[suspected_anomalies] = float("nan")
+    return linear(output.to_frame(), inplace=False).squeeze()
+
+
+def build_hydro(eia_api_key, hydro_plants, year, **hydro_kwargs):
+    """Use plant-level data to build hydro profiles.
+
```

Comment:
The current function has the data downloading function and hourly profile processing codes mixed in one method. I wonder if we can extract the hourly profile codes into a separate function, like `produc_hourly_profile()`?  This makes the codes easier to understand and to improve the profile generation method in the future.

Reply:
Done.